lib/ansible/modules/cloud/ovirt/ovirt_snapshot.py

```diff
@@ -222,6 +236,23 @@ def restore_snapshot(module, vm_service, snapshots_service):
     }
 
 
+def remove_old_snapshosts(module, snapshots_service):
+    from datetime import datetime
+    import time
+    deleted_snapshot = []
+    changed = False
+    date_now = datetime.now()
+    for snapshot in snapshots_service.list():
+        if snapshot.vm and snapshot.vm.name == module.params.get('vm_name'):
```

Comment:
I think it should be just fine to check `if snapshot.vm is not None` because you are already using `snapshots_service` of that vm, so you don't need to check if names are same.